### PR TITLE
Change the BMH's scope from Namespaced to Cluster scope

### PR DESCRIPTION
**Pull request by sadasu**

### CODE REVIEW

**Comment by dhellmann** [comment]
This is a pretty significant change to the API. I think we'll want a design doc explaining why we need to make it over in metal3-docs before we make the code change.

**Comment by s3rj1k** [comment]
@dhellmann Can this be configurable some how?

**Comment by s3rj1k** [comment]
How about creating BMHClustered? And supporting both cases.
Also it would be hard migrating current namespaced BMHs to cluster scope in production.

File: config/crd/bases/metal3.io_baremetalhosts.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.4.1
+    controller-gen.kubebuilder.io/version: v0.4.0
```
Comment (dtantsur) [inline config/crd/bases/metal3.io_baremetalhosts.yaml:7]: Why a version downgrade?

Comment (kirankt) [reply config/crd/bases/metal3.io_baremetalhosts.yaml:7]: My guess is that the local controller-gen version is different. Could be remedied easily.

**Comment by sadasu** [comment]
After further investigation if appears that this change is not needed and the wrong direction for what I was looking for. Thanks for taking a look and providing comments.

**Pull request closed without merging**